team-validator.js

```diff
@@ -22,7 +22,11 @@ class Validator {
 		if (supplementaryBanlist && supplementaryBanlist.length) {
 			format = Object.assign({}, format);
 			if (format.banlistTable) delete format.banlistTable;
-			if (!format.banlist) format.banlist = [];
+			if (format.banlist) {
+				format.banlist = format.banlist.slice(0);
```

Comment:
I see. Can this still be merged for now? I figure it will be easy enough to update when the refactor is fully decided.